nixos/modules/config/defaults.nix

```diff
@@ -0,0 +1,68 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.users.defaults;
+in {
+  options.users.defaults = {
+    editor = mkOption {
+      type = types.editorPackage;
+      default = pkgs.nano;
+      defaultText = literalExample "pkgs.nano";
+      example = literalExample "pkgs.kakoune";
+      description = ''
+        The package to use as the default editor.
+
+        To change the <varname>$EDITOR</varname> variable, set
+        the editorCommand passthru of the package. For example:
+
+        <programlisting>
+        users.defaults.editor = pkgs.vim // { editorCommand = "gvim -f"; };
```

Comment:
Should we encourage users to open a bug report if that attribute is missing?

Reply:
In the options documentation?

Reply:
I am not sure! I was just thinking about that when I saw the suggested ad-hoc set merging,.

Reply:
Do you feel the description isn't clear enough about that being the way to *override* the attribute instead of merely adding it? I could perhaps reword that a bit if you feel it's ambiguous.

Reply:
maybe something like this:
```
   description = ''
        The package to use as the default editor.

       To change the <varname>$EDITOR</varname> variable, most
       editors will be configured enough that, for example:
 <programlisting>
        users.defaults.editor = pkgs.vim;
</programlisting>
will be enough, but if you want to override what the package provides as <varname>editorCommand</varname>:

         set the editorCommand passthru of the package. For example:
        <programlisting>
        users.defaults.editor = pkgs.vim // { editorCommand = "gvim -f"; };
       </programlisting>
'';
```